src/test/java/com/williamfeliciano/sba_movies/services/MoviesServiceTests.java

```diff
@@ -0,0 +1,87 @@
+package com.williamfeliciano.sba_movies.services;
+
+import com.williamfeliciano.sba_movies.dtos.MovieDetailsDto;
+import com.williamfeliciano.sba_movies.dtos.MovieDto;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SpringBootTest
+public class MoviesServiceTests {
+
+    @Mock
+    private static MoviesService moviesService;
+    private static List<MovieDto> allMoviesMock;
+
+    private static MovieDetailsDto movie1;
+
+    @BeforeAll
+    public static void setUp() {
+        movie1 = MovieDetailsDto.builder()
+                .id(1L)
+                .title("The Shawshank Redemption")
+                .description("Two imprisoned men bond over a number of years, finding solace and eventual redemption through acts of common decency.")
+                .year(1994)
+                .rating(5)
+                .directors(List.of("Frank Darabont"))
+                .actors(List.of("Tim Robbins", "Morgan Freeman"))
+                .genres(List.of("Drama"))
+                .build();
+
+        MovieDetailsDto movie2 = MovieDetailsDto.builder()
+                .id(2L)
+                .title("The Godfather")
+                .description("An organized crime dynasty's aging patriarch transfers control of his clandestine empire to his reluctant son.")
+                .year(1972)
+                .rating(5)
+                .directors(List.of("Francis Ford Coppola"))
+                .actors(List.of("Marlon Brando", "Al Pacino"))
+                .genres(List.of("Crime", "Drama"))
+                .build();
+
+        List<MovieDetailsDto> movies = List.of(movie1, movie2);
+        allMoviesMock = List.of(
+                MovieDto.builder()
+                        .id(movie1.getId())
+                        .title(movie1.getTitle())
+                        .rating(movie1.getRating())
+                        .year(movie1.getYear())
+                        .build(),
+                MovieDto.builder()
+                        .id(movie2.getId())
+                        .title(movie2.getTitle())
+                        .rating(movie2.getRating())
+                        .year(movie2.getYear())
+                        .build()
+        );
+
+        moviesService = new MoviesService(movies);
+
+    }
+
+    @Test
+    public void getAllMoviesTest(){
+        Mockito.when(moviesService.getAllMovies())
```

Comment:
If `MoviesService` is the class you want to test, you can't use Mockito to return an expected value.
This line indicates that when calling the method `getAllMovies`, it will always return `allMoviesMock`. Nothing inside the `MoviesService` will be called.
Once you set the hardcoded list of movies in the controller, in the `setUp` method, just call `getAllMovies` and check the returned values.